src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -90,6 +92,12 @@ public ProAI(final String name, final String type) {
     storedPurchaseTerritories = null;
     storedPoliticalActions = null;
     storedStrafingTerritories = new ArrayList<Territory>();
+    initializeBattleCalculator();
+  }
+
+  public void initializeBattleCalculator() {
+    final IOddsCalculator calc = new ConcurrentOddsCalculator("ProAI");
+    ProBattleUtils.setOddsCalculator(calc);
```

Comment:
Usage of static state here is *really* procedural. Plus seems error prone too. If you mix ProAI and FastAI players in a game, does the last one win? Does  'initializeBattleCalculator' get call redundantly and/or before every time we call the battle calculator?

My suggestion for how to fix:  extract all of the OddsCalculation stuff out of ProBattleUtils, create an AIOddsCalculator class. Then inject the new AIOddsCalculator into your AI classes (constructor param).